document/core/exec/runtime.rst

```diff
@@ -535,21 +532,34 @@ and a reference to the function's own :ref:`module instance <syntax-moduleinst>`
 
 The values of the locals are mutated by respective :ref:`variable instructions <syntax-instr-variable>`.
 
+.. _syntax-handler:
+
 Exception handlers
 ..................
 
-Like labels, exception handlers carry the return arity :math:`n` of the
-respective |TRY| block, and their associated branch *target*, which is
-expressed syntactically as an :ref:`instruction <syntax-instr>` sequence:
+Exception handlers are installed by |TRY| instructions and are either *catching handlers* or *delegating handlers*.
+
+Catching handlers start with the identifier |CATCHadm| and carry a mapping from :ref:`tag addresses <syntax-tagaddr>`
+to their associated branch *targets*, each of which is expressed syntactically as a possibly empty sequence of
+:ref:`instructions <syntax-instr>` possibly following a :ref:`tag address <syntax-tagaddr>`.
+If there is no :ref:`tag address <syntax-tagaddr>`, the instructions of that target correspond to a |CATCHALL| clause.
+
+.. todo::
+   Add prose for delegating handlers.
 
 .. math::
-   \begin{array}{llll}
-   \production{(handler)} & \X{handler} &::=&
-     \CATCH_n\{\instr^\ast\}\\
+   \begin{array}{llllll}
+     \production{(handler)} & \handler &::=& \CATCHadm\{\tagaddr^?~\instr^\ast\}^\ast &|& \DELEGATEadm\{l\}
    \end{array}
 
-Intuitively, :math:`\instr^\ast` is the *continuation* to execute
-when the handler catches a thrown exception.
+Intuitively, for each target :math:`\{\tagaddr^?~\instr^\ast\}` of a |CATCHadm|, :math:`\instr^\ast` is the *continuation* to execute
+when the handler catches a thrown exception with tag |tagaddr|, or for any exception, when a target specifies no tag address.
+In that case, we say that the exception is handled by the exception handler |CATCHadm|.
```

Comment:
As a stylistic remark, the spec avoids formulations in first person like using "we". (Although I noticed that some recently sneaked in through the SIMD extension.)